### PR TITLE
Set remaining mypy setting to true

### LINKED ISSUE #101: Enable stricter mypy checks
## What was wrong?

Missing and/or wrong type definitions.

## How can it be fixed?

All of the strictness checks in mypy.ini should be set to `True`, like:
https://github.com/ethereum/ethereum-python-project-template/blob/582e9aa475bb56befb9a56188d2d33d176442b19/mypy.ini

There is a decent amount of work to do to add type annotations. There are a few wrong definitions too. Fix those up, and then the stricter checks will enforce it going forward.

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Tighten up typing!

Closes #101

### How was it fixed?

Set `mypy` rules `disallow_any_generics`, `disallow_subclassing_any` and `disallow_untyped_decorators` to `true`, fix errors

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-account/blob/main/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

File: eth_account/_utils/transaction_utils.py [hunk 1]
```diff
@@ -104,7 +107,9 @@ def transaction_rlp_to_rpc_structure(dictionary: Dict[str, Any]) -> Dict[str, An
     return dictionary
 
 
-def _access_list_rlp_to_rpc_structure(access_list: Sequence) -> Sequence:
+def _access_list_rlp_to_rpc_structure(
+    access_list: Sequence[Any],
```
Comment (kclowes) [inline eth_account/_utils/transaction_utils.py]: It looks like this is a pretty specific structure based on `t[0]` and `t[1]` from line 117. Looks like a two-tuple with address as the first and storage keys as the second. Maybe something like:
```suggestion
    access_list: Tuple[Address, Sequence[Bytes32]]
```

File: eth_account/_utils/transaction_utils.py [hunk 2]
```diff
@@ -104,7 +107,9 @@ def transaction_rlp_to_rpc_structure(dictionary: Dict[str, Any]) -> Dict[str, An
...
+) -> Sequence[Dict[str, Any]]:
```
Comment (kclowes) [inline eth_account/_utils/transaction_utils.py]: I think this can just be `AccessList`

**Review by kclowes — COMMENTED** [review]
I think we can use the shiny new `AccessList` here. Left a couple inline comments, but looks good otherwise!

File: eth_account/_utils/transaction_utils.py (hunk 1 above)
Comment (pacrob) [reply eth_account/_utils/transaction_utils.py]: We're mostly using `HexStr` for addresses here. I'll be taking another look at all this to see if the `TxData` typed dict from web3 could be shared with eth-account too.

**Review by kclowes — APPROVED** [review]
LGTM! 💥

File: eth_account/types.py [hunk 3]
```diff
@@ -19,5 +20,6 @@
 PrivateKeyType = Union[Bytes32, int, HexStr, PrivateKey]
 
 AccessList = Sequence[Dict[str, Union[HexStr, Sequence[HexStr]]]]
+RLPStructuredAccessList = Tuple[Tuple[HexStr, Tuple[HexStr, ...]], ...]
```
Comment (fselmo) [inline eth_account/types.py]: I wonder if these should be `Sequence` as well? Rather than `Tuple`

Comment (pacrob) [reply eth_account/types.py]: updated

File: eth_account/types.py [hunk 4]
```diff
@@ -19,5 +20,6 @@
 PrivateKeyType = Union[Bytes32, int, HexStr, PrivateKey]
 
 AccessList = Sequence[Dict[str, Union[HexStr, Sequence[HexStr]]]]
+RLPStructuredAccessList = Sequence[Tuple[HexStr, Tuple[HexStr, ...]]]
```
Comment (fselmo) [inline eth_account/types.py]: Let's use `Sequence` here as well

```suggestion
RLPStructuredAccessList = Sequence[Sequence[HexStr, Sequence[HexStr, ...]]]
```

Comment (pacrob) [reply eth_account/types.py]: updated

**Review by fselmo — APPROVED** [review]
lgtm!

**Pull request merged**